### PR TITLE
Fixing some issues with coco split utility

### PR DESCRIPTION
**Pull request by nasheedyasin**
Fixing the issue with the coco split utility failing when info and license fields aren't available.

### CODE REVIEW

**Comment by lolipopshock** [comment]
Thank you very much! I will check it ASAP!

File: utils/cocosplit.py [hunk 1]
```diff
@@ -6,19 +6,18 @@
...
 parser = argparse.ArgumentParser(description='Splits COCO annotations file into training and test sets.')
-parser.add_argument('annotations', metavar='coco_annotations', type=str,
+parser.add_argument('--annotation_path', metavar='coco_annotations', type=str,
                     help='Path to COCO annotations file.')
-parser.add_argument('train', type=str, help='Where to store COCO training annotations')
-parser.add_argument('test', type=str, help='Where to store COCO test annotations')
-parser.add_argument('-s', dest='split_ratio', type=float, required=True,
+parser.add_argument('--train', type=str, help='Where to store COCO training annotations')
+parser.add_argument('--test', type=str, help='Where to store COCO test annotations')
+parser.add_argument('--s', dest='split_ratio', type=float, required=True,
```
Comment (lolipopshock) [inline utils/cocosplit.py:13]: Thanks for the change - could we remove one `-` for the abbreviated name? or just use `--split-ratio`.

File: utils/cocosplit.py [hunk 2]
```diff
@@ -33,25 +32,48 @@ def main(annotation_path,
...
+        else:
+            # Merging the 2 image lists (i.e. with and without annotation)
+            tr_ann.extend(tr_wo_ann)
+            ts_ann.extend(ts_wo_ann)
+
+            tr, ts = tr_ann, ts_ann
 
-        x, y = train_test_split(images, train_size=split_ratio, random_state=random_state)
+        # Train Data
+        coco.update({'images': tr,
```
Comment (lolipopshock) [inline utils/cocosplit.py:64]: Clever way for handling the metadata fields!

File: tools/train_net.py [hunk 3]
```diff
@@ -114,6 +147,9 @@ def main(args):
         pd.DataFrame(res).to_csv(f'{cfg.OUTPUT_DIR}/eval.csv')
         return res
 
+    # Ensure that the Output directory exists
+    os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
```
Comment (lolipopshock) [inline tools/train_net.py:151]: I think Detectron2 will automatically handle the creation of the new directories. But good to add!

File: tools/train_net.py [hunk 4]
```diff
@@ -135,22 +171,22 @@ def main(args):
     parser = default_argument_parser()
 
     # Extra Configurations for dataset names and paths
```
Comment (lolipopshock) [inline tools/train_net.py:173]: Could we keep the spacing before `default` to make the configs visually recognizable? I think you could add `fmt:off` and `fmt:on` before and after the parser specification to disable the auto-listing. Thanks!

**Review by lolipopshock — CHANGES_REQUESTED** [review]
Great work! I've added a few comments on some details - other than that it looks good to me!

File: utils/cocosplit.py [hunk 5]
```diff
@@ -33,25 +32,48 @@ def main(annotation_path,
 
     with open(annotation_path, 'rt', encoding='UTF-8') as annotations:
         coco = json.load(annotations)
```
[the remark this answers is not in the document]
Comment (nasheedyasin) [reply utils/cocosplit.py:34]: Ah yes, I think this slipped my mind.

File: utils/cocosplit.py (hunk 2 above)
Comment (nasheedyasin) [reply utils/cocosplit.py:64]: Haha, thanks :)

**Comment by nasheedyasin** [comment]
Hey, have added the changes as suggested :)
Thanks for the encouragement xD My first open src PR

**Review by lolipopshock — APPROVED** [review]
LGTM! Thanks!

File: tools/train_net.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def get_augs(cfg):
     augs.append(T.RandomFlip(horizontal=horizontal_flip,
                              vertical=not horizontal_flip))
     # Rotate the image between -90 to 0 degrees clockwise around the centre
-    augs.append(T.RandomRotation(angle=[90.0, 0.0]))
+    augs.append(T.RandomRotation(angle=[-90.0, 0.0]))
```
[the remark this answers is not in the document]
Comment (nasheedyasin) [reply tools/train_net.py:49]: Well, the I misread the detectron2 documentation and thought that the rotation was anti-clockwise about the center. It was in fact clockwise.

[-90, 90] could be a good idea, but it really depends on the data that the trainer has got. In my case, [-90, 0] sufficed as there was a really low probability of encountering documents that were rotated clockwise 90 degrees. The value there should be basically a placeholder that the use could fill.

I will try to see if there is a way to allow the user to choose this range from the config.yaml file, but so far I have been unable to find a non-code way to do it.

Also I consciously avoided having a 180 degree range as I noticed that having upside down text was bad for the model when working on textual images.

[the remark this answers is not in the document]
Comment (lolipopshock) [reply tools/train_net.py:49]: Gotcha - thanks! And I agree with you - document images are super sensitive to the orientation of the training data - in one earlier experiment, even a horizontal flip augmentation can decrease the model performance. And probably the rotation augmentation is unnecessary in certain cases (for example, for newspaper images, where almost all the newspaper are positioned in a certain way).

**Pull request merged**